runtime/common/src/main/kotlin/kotlinx/serialization/json/serializers/JsonElementSerializer.kt

```diff
@@ -93,23 +147,25 @@ private object JsonPrimitiveSerializer : KSerializer<JsonPrimitive> {
         encoder.encodeString(obj.content)
     }
 
-    override fun deserialize(decoder: Decoder): JsonPrimitive {
-        return JsonPrimitive(decoder.decodeString())
+    override fun deserialize(decoder: Decoder): JsonLiteral {
```

Comment:
Todo: how we can deserialize ints, floats and so on with preserving info about it?

Reply:
Not in this API shape.
We can't distinguish them because we are not doing so on the type system level.

Reply:
Yes